docs/style-spec/expressions.md

```diff
@@ -0,0 +1,40 @@
+# Style Function Expressions
+
+**Constants:**
+- LN2
+- PI
+- e
+
+**Literal:**
+- JSON string / number / boolean literal
+
+**Property lookup:**
+- Feature property:  `{ "ref": "feature", "key": propertyName, "default": defaultValue}`
+- Map property:  `{"ref": "map",  "key": "zoom"|"pitch"|… }`
+  - Note: expressions that refer to the map zoom level are only evaluated at integer zoomslevels. When the map is at non-integer zoom levels, the expression's value will be approximated using linear or exponential interpolation.
+
+**Decision:**
+- `["if", boolean_expr, expr_if_true, expr_if_false]` 
+  - Alternative: `[` `"``if``"``, [ filter_expr, expr_if_true ], [ filter_expr, expr_if_true], …, expr_otherwise]` (where the first matching filter wins)
```

Comment:
As with conditionals, I'm leaning back towards the first proposal here (the one that's equivalent to ternary operator).  Compactness in the JSON representation doesn't seem particularly important, since no one will actually be authoring expressions via AST (except for us when we write unit tests 😄 )